qiskit/pulse/schedule.py

```diff
@@ -274,6 +274,21 @@ def parameters(self) -> Set:
         """Parameters which determine the schedule behavior."""
         return self._parameter_manager.parameters
 
+    @property
+    def pulses(self):
+        """Get a list of all of the pulses used in this schedule."""
+        pulses = []
+        for instruction in self.instructions:
+            pulses.append(instruction[1].pulse)
```

Comment:
`instruction` could be non-`Play` type, i.e. there are many pulse instructions that don't take `pulse` as an operand. See below for more details.
https://qiskit.org/documentation/stable/0.24/stubs/qiskit.pulse.instructions.html#module-qiskit.pulse.instructions

Another question is about the handling of duplicated pulse. For example, in a calibration sequence such as error amplification you may get many duplicated pulses as they repeats the same gate. I think it's better to only return unique entries. Unfortunately `SymbolicPulse` instance is now unhashable and O1 look-up cannot be used here.

Reply:
so can I create a set and add instructions to it and finally convert it to list and return it?